route/src/route_generator_worker.cpp

```diff
@@ -469,12 +469,36 @@ namespace route {
         return true;
     }
 
+    void RouteGeneratorWorker::lookupFrontBumperTransform() 
+    {
+        tf2_listener_.reset(new tf2_ros::TransformListener(tf2_buffer_));
+        tf2_buffer_.setUsingDedicatedThread(true);
+        try
+        {
+            tf_ = tf2_buffer_.lookupTransform("map", "vehicle_front", ros::Time(0), ros::Duration(0.10)); //save to local copy of transform 20 sec timeout
```

Comment:
I think I understand now, my question from above. This is the transform from map -> vehicle front that's being queried, which is why it needs to be requeried every timestep, basically not using the pose topic data but instead using this transform value (that was computed by TF2 based on this transform).

I think what I was suggesting was more along the lines of just looking up the base_link -> vehicle_front transform once, and then applying that transform to the pose data at each timestep. I'm not sure what the overhead on a transform look up is, but if we do the transform look up once I think that's probably more efficient than doing it at 10+Hz.

Reply:
I understand what you mean, and that is what I initially tried. However, I realized the baselink->bumper transform is in the base link frame, and the pose is in the map frame. So they don't work together. For example, if the current pose = (3, 3, 0) then when  transformed, bumper pose = (7, 3, 0). However, the 4 m added to x, does not necessarily mean the pose is shifted longitudinally to the front, it could be any direction, which is incorrect and not to mention causes route generation to fail.

Reply:
Do the transforms themselves encode the rotation data as well? Such that the rotation encoded in the map -> base_link transform, the vehicle's orientation w.r.t. the map, should already be accounted for prior to us adding the front bumper offset into the mix. 

With map as our baseline, we have the 4x4 transform matrix describing the position and orientation of the vehicle rear axle reference w.r.t. to map frame T<sub>1</sub> and another 4x4 transform matrix describing the position and orientation of the front bumper reference point w.r.t. rear axle T<sub>2</sub>, I think the pose of the front bumper w.r.t to map frame should be T<sub>1</sub>T<sub>2</sub> right?